sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -484,15 +484,17 @@ object IntervalUtils {
     var pointPrefixed: Boolean = false
 
     def trimToNextState(b: Byte, next: ParseState): Unit = {
-      b match {
-        case ' ' => i += 1
-        case _ => state = next
+      if (b <= ' ') {
+        i += 1
+      } else {
+        state = next
       }
     }
 
     def currentWord: UTF8String = {
-      val strings = s.split(UTF8String.blankString(1), -1)
-      val lenRight = s.substring(i, s.numBytes()).split(UTF8String.blankString(1), -1).length
+      val sep = UTF8String.fromString("\\s+")
+      val strings = s.split(sep, -1)
```

Comment:
`UTF8String.split` simply calls java `String.split`. Since we only need to get the current word for error message, can we just call `String.split`?

Reply:
yea, now `String.split` shall work, let me change this.